pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -357,21 +358,36 @@ func (c *Cluster) addOrRemoveExternalMonitor(status client.MonStatusResponse) (b
 				monPort := cephutil.GetPortFromEndpoint(endpoint)
 				logger.Infof("new external mon %s found: %s, adding it", mon.Name, endpoint)
 				c.ClusterInfo.Monitors[mon.Name] = cephconfig.NewMonInfo(mon.Name, monIP, monPort)
-				changed = true
+			} else {
+				logger.Debugf("mon %s is not in quorum and not in ClusterInfo", mon.Name)
 			}
-			logger.Debugf("mon %s is not in quorum and not in ClusterInfo", mon.Name)
+			changed = true
```

Comment:
Now it feels strange to return True, it seems like it's always True since we re-populate clusterInfo all the time.
If that's the case, we should probably always call `c.saveMonConfig()` after `c.addOrRemoveExternalMonitor(status)` is called.

Reply:
No really, only if the mon from status is not found in oldClusterInfoMonitors.
If the monitor is found in oldClusterinfoMonitors, it will added to c.ClusterInfo.Monitors again